src/sync/PushManager/index.js

```diff
@@ -0,0 +1,201 @@
+
+    timeoutId = setTimeout(connectPush, delayInSeconds * 1000);
+  }
+
+  function connectPush() {
+    log.info('Connecting to push streaming.');
+
+    const userKeys = clientContexts ? Object.keys(clientContexts) : undefined;
+    authenticate(settings, userKeys).then(
+      function (authData) {
+        // restart backoff retry counter for auth and SSE connections, due to HTTP/network errors
+        reauthBackoff.reset();
+        sseReconnectBackoff.reset(); // reset backoff in case SSE conexion has opened after a HTTP or network error.
+
+        // emit PUSH_DISCONNECT if org is not whitelisted
+        if (!authData.pushEnabled) {
+          log.error('Streaming is not enabled for the organization. Switching to polling mode.');
+          pushEmitter.emit(PUSH_DISCONNECT); // there is no need to close sseClient (it is not open on this scenario)
+          return;
+        }
+
+        // don't open SSE connection if a new shared client was added, since it means that a new authentication is taking place
+        if (userKeys && userKeys.length < Object.keys(clientContexts).length) {
+          return;
+        }
+
+        // Connect to SSE and schedule refresh token
+        const decodedToken = authData.decodedToken;
+        sseClient.open(authData);
+        scheduleTokenRefresh(decodedToken.iat, decodedToken.exp);
+      }
+    ).catch(
+      function (error) {
+
+        sseClient.close(); // no harm if already disconnected
+        pushEmitter.emit(PUSH_DISCONNECT); // no harm if `PUSH_DISCONNECT` was already notified
+
+        if (error.statusCode) {
+          switch (error.statusCode) {
+            case 401: // invalid api key
+              log.error(`Fail to authenticate for push notifications, with error message: "${error.message}".`);
+              return;
+          }
+        }
+
+        // Branch for other HTTP and network errors
+        const delayInMillis = reauthBackoff.scheduleCall();
+        log.error(`Fail to authenticate for push notifications, with error message: "${error.message}". Attempting to reauthenticate in ${delayInMillis / 1000} seconds.`);
+      }
+    );
+  }
+
+  function disconnectPush() {
+    sseClient.close();
+
+    // cancel timeouts if previously established
+    if (timeoutId) clearTimeout(timeoutId);
+    reauthBackoff.reset();
+    sseReconnectBackoff.reset();
+  }
+
+  /** Fallbacking due to STREAMING_DISABLED control event */
+
+  pushEmitter.on(PUSH_DISABLED, function () {
+    disconnectPush();
+    pushEmitter.emit(PUSH_DISCONNECT); // no harm if polling already
+  });
+
+  /** Fallbacking due to SSE errors */
+
+  pushEmitter.on(SSE_ERROR, function (error) { // HTTP or network error in SSE connection
+    // SSE connection is closed to avoid repeated errors due to retries
+    sseClient.close();
+
+    // retries are handled via backoff algorithm
+    let delayInMillis = (error.parsedData && (error.parsedData.statusCode === 400 || error.parsedData.statusCode === 401)) ?
```

Comment:
Assign the code to a variable to make the ternary more readable. If you want to keep it as is, with long conditions might be best to use an if statement directly so the two flows are clear.